src/mono/mono/metadata/icall-eventpipe.c

```diff
@@ -491,6 +522,101 @@ eventpipe_execute_rundown (
 	return TRUE;
 }
 
+static
+gboolean
+eventpipe_walk_managed_stack_for_thread_func (
+	MonoStackFrameInfo *frame,
+	MonoContext *ctx,
+	gpointer data)
+{
+	g_assert_checked (frame != NULL);
+	g_assert_checked (data != NULL);
+
+	switch (frame->type) {
+	case FRAME_TYPE_DEBUGGER_INVOKE:
+	case FRAME_TYPE_MANAGED_TO_NATIVE:
+	case FRAME_TYPE_TRAMPOLINE:
+	case FRAME_TYPE_INTERP_TO_MANAGED:
+	case FRAME_TYPE_INTERP_TO_MANAGED_WITH_CTX:
+		return FALSE;
+	case FRAME_TYPE_MANAGED:
+	case FRAME_TYPE_INTERP:
+		if (!frame->ji)
+			return FALSE;
+		MonoMethod *method = frame->ji->async ? NULL : frame->actual_method;
+		ep_stack_contents_append ((EventPipeStackContents *)data, (uintptr_t)((uint8_t*)frame->ji->code_start + frame->native_offset), method);
+		return ep_stack_contents_get_length ((EventPipeStackContents *)data) >= EP_MAX_STACK_DEPTH;
+	default:
+		g_assert_not_reached ();
+		return FALSE;
+	}
+}
+
+static
+gboolean
+eventpipe_walk_managed_stack_for_thread (
+	ep_rt_thread_handle_t thread,
+	EventPipeStackContents *stack_contents)
+{
+	g_assert_checked (thread != NULL);
+	g_assert_checked (stack_contents != NULL);
+
```

Comment:
Ditto.

Reply:
Could change that since function is only called once per requested callstack.